.github/workflows/go.yml

```diff
@@ -13,7 +13,47 @@ env:
   go-cache-name: go
 
 jobs:
+
+  changedfiles:
+    runs-on: [ubuntu-latest]
+    outputs:
+      docs: ${{ steps.changes.outputs.docs }}
+      docs_changes: ${{ steps.changefiles.outputs.docs_changes}}
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - name: Get changed files
+      id: changes
+      run: |
+        echo "::set-output name=docs::$(git diff --name-only ${{ github.event.before }}..${{ github.event.after }} | xargs)"
+    - name: echo output
+      id: changefiles
+      run: |
+        docs_changes=false
+        for file in ${{ steps.changes.outputs.docs }}; do
+
+             echo "changed files are $file"
+             ext="${file##*.}"
+             echo "$ext"
+             echo "$docs_changes"
+            if [[ ($ext != "md") && ($file != "hack/.md_links_config.json") ]]; then
+                  docs_changes=true
+                  break
+            fi
+        done
+        if $docs_changes; then
+          echo "Setting 'docs_changes' to 'yes'"
+          echo "::set-output name=docs_changes::yes"
```

Comment:
I suppose this `docs_changes` means changes happen in non docs files? maybe better to name it as `code_changes`